lib/twilio/call.ts

```diff
@@ -1204,7 +1206,10 @@ class Call extends EventEmitter {
 
     this._log.info('Received HANGUP from gateway');
     if (payload.error) {
-      const error = new GeneralErrors.ConnectionError('Error sent from gateway in HANGUP');
+      const code = payload.error.code;
+      const error = typeof code === 'number' && hasErrorByCode(code)
+        ? new (getErrorByCode(code))(payload.error.message)
+        : new GeneralErrors.ConnectionError('Error sent from gateway in HANGUP');
```

Comment:
Do we still publish hangup errors via the device object? If so, should we also update device? If not, can we update the changelog to not mention device?

Reply:
We don't publish to insights from the device object at all, it seems. Unless I'm missing something. And the Call object publishes all errors it emits to insights, and the hangup errors are emitted and therefore published.

Reply:
Sorry but the use of 'publish' might've caused confusion. I meant 'emit'.
Do we still emit hangup errors via the device object? If so, should we also update device? If not, can we update the changelog to not mention device?

Reply:
We do not catch and then re-emit hangup errors with the device, just the call. The device will catch and re-emit payload messages with type `error`, however. See my other comment for my concerns.